crates/iceberg/src/transaction/mod.rs

```diff
@@ -76,7 +76,7 @@ impl Transaction {
         Ok(())
     }
 
-    fn apply(
+    pub(crate) fn apply(
```

Comment:
With the new Transaction API, Transaction should only hold `actions` (pending actions to commit), `updates` and `requirements` should be removed after this PR: https://github.com/apache/iceberg-rust/pull/1451

Based on this thought, `tx.apply` shouldn't be accessible to anything except `tx.commit`